ddl/partition.go

```diff
@@ -361,6 +372,32 @@ func buildHashPartitionDefinitions(ctx sessionctx.Context, s *ast.CreateTableStm
 	return nil
 }
 
+func buildListPartitionDefinitions(s *ast.CreateTableStmt, pi *model.PartitionInfo) (err error) {
+	for _, def := range s.Partition.Definitions {
+		comment, _ := def.Comment()
+		err = checkTooLongTable(def.Name)
+		if err != nil {
+			return err
+		}
+		piDef := model.PartitionDefinition{
+			Name:    def.Name,
+			Comment: comment,
+		}
+
+		buf := new(bytes.Buffer)
+		for _, vs := range def.Clause.(*ast.PartitionDefinitionClauseIn).Values {
+			if len(vs) != 1 {
+				return fmt.Errorf("not support list columns partition")
+			}
+			vs[0].Format(buf)
+			piDef.InValues = append(piDef.InValues, buf.String())
```

Comment:
> The persistent data is a string of integer, rather than an integer?

Yes, since the value also could be `null`, here is an example:
```sql
create table t1 (id int) partition by list  (id) (partition p0 values in (1,null));
```